### PR TITLE
Added a new getToken method to Tokens which will return the token obj…

### PR DESCRIPTION
**Pull request by cratter49**
…ect that is associated with the given apiToken string.

### CODE REVIEW

File: lib/model/Tokens.js [hunk 1]
```diff
@@ -95,6 +95,34 @@ Tokens.prototype.connectByApiToken = function(tenantName, apiToken, callback) {
...
 
+/**
+ * Authenticate with the API endpoint to get the entire token object
+ * 
+ * @param {string} tenantName - name of the tenant we want to access
+ * @param {string} apiToken - the apiToken retrieved from mPulse UI
+ * @param {function} callback - 
+ * [Called after the API Auth request was finalized]{@link Tokens~getToken(callback)}
+ */
+Tokens.prototype.getToken = function(tenantName, apiToken, callback) {
```
Comment (msolnit) [inline lib/model/Tokens.js]: @cratter49 This isn't an API token, so it shouldn't be called `apiToken`.  Just `token`.

File: lib/model/Tokens.js [hunk 2]
```diff
@@ -95,6 +95,34 @@ Tokens.prototype.connectByApiToken = function(tenantName, apiToken, callback) {
     });
 }
 
+/**
+ * Authenticate with the API endpoint to get the entire token object
+ *
```
Comment (msolnit) [inline lib/model/Tokens.js]: @cratter49 This isn't actually authenticating, from what I can see. Can you please correct the JS Doc?

File: lib/model/Tokens.js (hunk 1 above)
Comment (cratter49) [reply lib/model/Tokens.js]: @msolnit Well in this case it's both an API - Authentication token being used to make a request to the server in addition to being used as the ID of the object to find.

Comment (msolnit) [reply lib/model/Tokens.js]: No it's not.  This method will only work after you've already authenticated via `connect()`, right?

File: lib/model/Tokens.js (hunk 2 above)
Comment (bluesmoon) [reply lib/model/Tokens.js]: This appears to be the "fetch token details" api endpoint that is conveniently omitted from the docs here: http://docs.soasta.com/repository-api/#tokens

Might be a good idea to update the API docs as well.

File: lib/model/Tokens.js (hunk 1 above)
Comment (bluesmoon) [reply lib/model/Tokens.js]: neither `tenantName` nor `apiToken` are required (or have any meaning) for this call, but `token` (ie, the thing returned by `connect*` is required.

File: lib/model/Tokens.js [hunk 3]
```diff
@@ -95,6 +95,34 @@ Tokens.prototype.connectByApiToken = function(tenantName, apiToken, callback) {
     });
 }
 
+/**
+ * Authenticate with the API endpoint to get the entire token object
+ * 
+ * @param {string} tenantName - name of the tenant we want to access
+ * @param {string} apiToken - the apiToken retrieved from mPulse UI
+ * @param {function} callback - 
+ * [Called after the API Auth request was finalized]{@link Tokens~getToken(callback)}
```
Comment (bluesmoon) [inline lib/model/Tokens.js]: I couldn't find the docs for `Tokens~getToken(callback)` and it's not clear from these docs how the callback should be defined or what it should expect.

**Pull request merged**